Adding a note about the updated release process (#4064)
flutter/website: src/docs/development/tools/sdk/releases.md

The {{site.sdk.channel | capitalize }} channel contains the
most stable Flutter builds. See [Flutter’s channels][] for details.

{{site.alert.note}}

>   As of release 1.17 (stable), we have updated our release process.

Reviewer: ```suggestion
  As of release 1.17 (stable), we've updated our release process,
```

>   This includes how we name the releases and their corresponding files.

Reviewer: ```suggestion
  including changes to the names of releases and their download files.
```

Reviewer 2: "how we name" makes more sense in this context to me.  if we want to be more specific we can say "we've changed our versioning schema," but I think "how we name" makes sense.

Reviewer: Maybe:

...changes to how we name releases and their download files.

Reviewer: I was trying to avoid too much "we" and unadorned "This". Although given the name of the link I didn't avoid "we" entirely.

>   For more information, see the "Updating our release process" section

Reviewer: ```suggestion
  For details, see the "Updating our release process" section
```
